sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -213,6 +213,20 @@ case class DataSourceResolution(
           }
           ShowTablesCommand(Some(namespace.quoted), pattern)
       }
+
+    case SetCatalogAndNamespaceStatement(catalogName, namespace) =>
+      catalogName match {
+        case Some(c) =>
+          validateCatalog(c)
+          SetCatalogAndNamespace(catalogManager, Some(c), namespace)
+        case None =>
+          assert(namespace.nonEmpty)
+          val CurrentCatalogAndNamespace(catalog, ns) = namespace.get
+          if (!catalog.asNamespaceCatalog.namespaceExists(ns.toArray)) {
+            throw new AnalysisException(s"Namespace '${ns.quoted}' not found")
```

Comment:
there is a `NoSuchNamespaceException`, shall we use that?

Reply:
changed. thanks.